packages/web-components/fast-foundation/src/data-grid/data-grid.ts

```diff
@@ -520,6 +585,175 @@ export class DataGrid extends FoundationElement {
                     );
                 }
                 break;
+
+            case "a":
+                if (!e.ctrlKey) {
+                    return;
+                }
+                switch (this.selectionMode) {
+                    case "multiRow":
+                        this.selectAllRows();
+                        e.preventDefault();
+                        return;
+                }
+                break;
+        }
+    }
+
+    public handleRowSelectedChange(e: CustomEvent): void {
+        if (e.defaultPrevented || this.selectionMode === "none") {
+            return;
+        }
+
+        const path: EventTarget[] = e.composedPath();
+        const rowMatch: EventTarget | undefined = path.find((target: EventTarget) => {
+            return this.rowElements.indexOf(target as HTMLElement) !== -1;
+        });
+
+        if (rowMatch) {
+            e.preventDefault();
+            const changedRow: DataGridRow = rowMatch as DataGridRow;
+            switch (this.selectionMode) {
+                case "singleRow":
+                    this.handleSingleRowSelection(changedRow);
+                    this.$emit("selectionchanged");
+                    break;
+
+                case "multiRow":
```

Comment:
Multi row selection is not working for me.

Reply:
Is your caps lock down?  I think I need to ignore case here.

Reply:
fixed some issues in multi-select related to confusing the index of the index in the array of indexes with the index in the array of indexes...  Should be good now.